tests/share/disambiguation/test_creativework.py

```diff
@@ -259,3 +260,96 @@ def test_no_timetraveling_many(self, Graph):
 
         older_graph.process()
         assert older_graph.nodes[0].change == {'description': 'The final description'}
+
+    def test_cannot_overwrite_cannonical(self, Graph):
```

Comment:
Could you add a test for canonical overwrites not canonical so `canonical=False` then `canonical=True`. That way all cases are covered.